src/Strategy/Cocoapods/PodfileLock.hs

```diff
@@ -114,99 +125,50 @@ data Remote = Remote
-dependenciesSectionParser = sectionParser "DEPENDENCIES:" DependencySection depParser
-
-specRepoParser :: Parser Section
-specRepoParser = sectionParser "SPEC REPOS:" SpecRepos remoteParser
-
-externalSourcesParser :: Parser Section
-externalSourcesParser = sectionParser "EXTERNAL SOURCES:" ExternalSources externalDepsParser
-
-checkoutOptionsParser :: Parser Section
-checkoutOptionsParser = sectionParser "CHECKOUT OPTIONS:" CheckoutOptions externalDepsParser
-
-sectionParser :: Text -> ([a] -> Section) -> Parser a -> Parser Section
-sectionParser sectionName lambda parser = nonIndented $
-  indentBlock $ do
-    _ <- chunk sectionName
-    pure (L.IndentMany Nothing (pure . lambda) parser)
-
-externalDepsParser :: Parser SourceDep
-externalDepsParser = indentBlock $ do
-  name <- lexeme (takeWhileP (Just "external dep parser") (/= ':'))
-  _ <- restOfLine
-  pure (L.IndentMany Nothing (pure . SourceDep name . Map.fromList) tagParser)
-
-tagParser :: Parser (Text, Text)
-tagParser = do
-  _ <- chunk ":"
-  tag <- lexeme (takeWhileP (Just "tag parser") (/= ':'))
-  _ <- chunk ": "
-  value <- restOfLine
-  pure (tag, value)
-
-remoteParser :: Parser Remote
-remoteParser = indentBlock $ do
-  location <- restOfLine
-  pure (L.IndentMany Nothing (pure . Remote (Text.dropWhileEnd (== ':') location)) depParser)
-
-podParser :: Parser Pod
-podParser = indentBlock $ do
-  name <- symbol "-" *> ignoreDoubleQuote *> findDep
-  version <- findVersion
-  _ <- restOfLine
-  pure (L.IndentMany Nothing (pure . Pod name version) depParser)
-
-depParser :: Parser Dep
-depParser = do
-  name <- symbol "-" *> ignoreDoubleQuote *> findDep
-  _ <- restOfLine
-  pure $ Dep name
-
-findDep :: Parser Text
-findDep = lexeme (takeWhile1P (Just "dep") isNotDoubleQuoteAndSpace)
-  where
-    isNotDoubleQuoteAndSpace :: Char -> Bool
-    isNotDoubleQuoteAndSpace c = c /= '"' && (not . C.isSpace) c
-
-findVersion :: Parser Text
-findVersion = between (char '(') (char ')') (lexeme (takeWhileP (Just "version") (/= ')')))
-
-restOfLine :: Parser Text
-restOfLine = takeWhileP (Just "ignored") (not . isEndLine)
-
-isEndLine :: Char -> Bool
-isEndLine '\n' = True
-isEndLine '\r' = True
-isEndLine _ = False
-
-nonIndented :: Parser a -> Parser a
-nonIndented = L.nonIndented scn
-
-indentBlock :: Parser (L.IndentOpt Parser a b) -> Parser a
-indentBlock = L.indentBlock scn
-
-scn :: Parser ()
-scn = L.space space1 empty empty
+parseName :: Parser Text
+parseName = lexeme $ takeWhile1P (Just "dep") (not . C.isSpace)
+
+parseVersion :: Parser Text
+parseVersion = between (char '(') (char ')') $ lexeme (takeWhileP (Just "version") (/= ')'))
+
+parseNameAndVersion :: Parser (Text, Text)
+parseNameAndVersion = do
+  name <- parseName
+  version <- parseVersion
+  pure (name, version)
+
+instance FromJSON PodLock where
+  parseJSON = Yaml.withObject "Podfile.lock content" $ \o -> do
+    pods <- o .: "PODS"
+    deps <- o .: "DEPENDENCIES"
+    pure $ PodLock pods deps
```

Comment:
Since this is a parser, we should prefer applicative parsing to monadic parsing:
```suggestion
instance FromJSON PodLock where
  parseJSON = Yaml.withObject "PodLock" $ \obj ->
    PodLock <$> obj .: "PODS"
      <*> obj .: "DEPENDENCIES"
```

Two nits as well:

- Prefer `obj` name to `o`.  It's codebase convention for almost every json and yaml parser we have.
- For the error string, it's sufficient and conventional to just use the name of the data structure.